Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1078,15 +1078,16 @@ public void UpdateLastBlockSyncedHeight(Wallet wallet, ChainedBlock chainedBlock
         /// <param name="chainCode">The chain code.</param>
         /// <param name="creationTime">The time this wallet was created.</param>
         /// <returns>The wallet object that was saved into the file system.</returns>
-        /// <exception cref="System.NotSupportedException"></exception>
+        /// <exception cref="WalletException">Thrown if wallet cannot be created.</exception>
         private Wallet GenerateWalletFile(string name, string encryptedSeed, byte[] chainCode, DateTimeOffset? creationTime = null)
         {
             Guard.NotEmpty(name, nameof(name));
             Guard.NotEmpty(encryptedSeed, nameof(encryptedSeed));
             Guard.NotNull(chainCode, nameof(chainCode));
             this.logger.LogTrace("({0}:'{1}')", nameof(name), name);
 
-            if (this.fileStorage.Exists($"{name}.{WalletFileExtension}"))
+            // Check if any wallet file already exists, with case insensitive comparison.
+            if (this.Wallets.Any(w => string.Compare(w.Name, name, true) == 0))
```

Comment:
it's probably better to use `w.Equals(name, StringComparison.OrdinalIgnoreCase);` for this.
or `string.equals(w.Name, name, StringComparison.OrdinalIgnoreCase)`
I'm basing myself on this 
![image](https://user-images.githubusercontent.com/1867877/31726783-bebd0a62-b41f-11e7-8662-8fec4cffec58.png)

https://docs.microsoft.com/en-us/dotnet/standard/base-types/comparing

Reply:
:+1: